proposals/0030-markdoc.md

```diff
@@ -0,0 +1,295 @@
+---
+
+<html lang="en">
+	<body>
+		<Content
+			config={{
+				variables: { underlineTitle: true },
+			}}
+			components={{
+				h1: Title,
+				marquee: Marquee,
+			}}
+		/>
+	</body>
+</html>
+```
+
+See [the detailed API section](#api) for information on all supported props, shared config, and advanced component mapping.
+
+# Background & Motivation
+
+Include any useful background detail that that explains why this RFC is important.
+What are the problems that this RFC sets out to solve? Why now? Be brief!
+
+It can be useful to illustrate your RFC as a user problem in this section.
+(ex: "Users have reported that it is difficult to do X in Astro today.")
+
+# Goals
+
+- **Create an `@astrojs/markdoc` integration** that adds `.mdoc` support to content collections.
+- **Support Astro components and server-rendered UI components** (React, Vue, Svelte, etc) within Markdoc files. Note this leaves client-rendered UI components out-of-scope (see non-goals).
+- **Benchmark Markdoc performance against Markdown and MDX** at 1k and 10k documents. This addresses problem (1) from the previous section.
+
+# Non-goals
+
+- Non-goal: **ESM import and `src/pages/` support** for Markdoc files. [See discussion](https://github.com/withastro/roadmap/discussions/478#discussioncomment-4975072) for context.
+- Out-of-scope: **Allowing the `.md` extension.** This would mean overriding Astro's `.md` renderer, which is tightly coupled to remark and your `markdown` configuration options today. We agree using `.md` for Markdoc is a common use case, and deserves a separate proposal to make Astro's Markdown rendering flexible.
+- Future: **A solution for client-rendered UI components.** Unlike MDX, Markdoc doesn't have a concept of directives, and our compiler doesn't have a clear way to dynamically render client-side components ([see challenges](https://gist.github.com/bholmesdev/491fa440efa8bd73d410d66c0c2143c2)). We will recommend users wrap their components in an Astro component to apply the `client:` directive.
+- Future: **Full alignment with Markdown and MDX** rendered result. Namely, the computed `headings` property (which can be tackled in future releases) and frontmatter manipulation via remark (since remark is incompatible with Markdoc).
+
+# Detailed Design
+
+## API
+
+Markdoc support will be specific to content collections. This means Markdoc files can be included alongside other content collection entries via the `.mdoc` extension. Entries can also be queried using the same `getCollection()` and `getEntryBySlug()` APIs for Markdown and MDX.
+
+### Rendering Markdoc content
+
+Users can render contents and configure Markdoc via the `<Content />` component. This will be exposed from the `render()` result, and feature two props:
+
+- `config?: import('@markdoc/markdoc').Config`: An (optional) Markdoc config to be used [during the transformation step](https://markdoc.dev/docs/render#transform). This includes configuration for Markdoc tags and variables.
```

Comment:
meta-point: this isn't a useful type annotation since we don't actually have that type anywhere accessible inside the RFC. To fix this, I'd add a code example similar to the one below that I ended up commenting on, but just to illustrate the basic config first.

Reply:
Hm, this type is based on the `@markdoc/markdoc` package and not the Astro internals, so it seemed sufficient to me. Would a hyperlink to the type def in the Markdoc repository be helpful instead of an annotation?